core/src/main/clojure/xtdb/util.clj

```diff
@@ -663,3 +664,14 @@
                           (cons x (step (rest s) (conj seen v)))))))
                   xs seen)))]
      (step coll #{}))))
+
+(defn max-direct-memory
+  "Returns the maximum direct memory supposed to be used by the system"
+  []
+  (if-let [size-string (get (System/getProperties) "MaxDirectMemory")]
+    (let [^long size (parse-long (subs size-string 0 (dec (count size-string))))]
+      (case (last size-string)
```

Comment:
nit: may need to lower-case this - I think you can specify it with upper-case too

Reply:
Btw this is a property we are specifying, not `-XX:MaxDirectMemorySize`.

Reply:
Just out of curiosity - Is this being used anywhere, currently?

Reply:
Yes, I use it on the AWS runs.